### PR TITLE
Only default two thirds/one-third layout when footer has two sections

### PR DESCRIPTION
**Pull request by nickcolley**
This approach is from: http://lea.verou.me/2011/01/styling-children-based-on-their-number-with-css3/

Changes the default layout of two-thirds/one-third only when there's two sections.

The first section can only be also the the 2nd last section (counted backwards) if there's two overall sections.

### Before
Before equal columns

### After
After equal columns

Improves https://github.com/alphagov/govuk-frontend/issues/1539

### CODE REVIEW

File: src/govuk/components/footer/_footer.scss [hunk 1]
```diff
@@ -196,9 +196,9 @@
     }
   }
 
-  // Sections two-third:one-third on desktop
+  // If the there are only two sections and set the layout to be two-third:one-third on desktop
```
Comment (ChrisBAshton) [inline src/govuk/components/footer/_footer.scss]:
```suggestion
  // If there are only two sections, set the layout to be two-third:one-third on desktop
```

File: src/govuk/components/footer/_footer.scss [hunk 2]
```diff
@@ -196,9 +196,9 @@
...
   @include mq ($from: desktop) {
-    .govuk-footer__section:first-child {
+    .govuk-footer__section:first-child:nth-last-child(2) {
```
Comment (ChrisBAshton) [inline src/govuk/components/footer/_footer.scss]: Won't this apply a `flex-grow: 2` to the 2nd-to-last child of the footer section _regardless_ of how many columns there are?

I.e. a three-column layout will set the middle column to `flex-grow: 2` - is that expected?

Comment (nickcolley) [reply src/govuk/components/footer/_footer.scss]: nth-last-child counts backwards, so it'll only target the first element and the 2nd from last element.

if you had three columns then the 2nd element would not also be the first element

Comment (ChrisBAshton) [reply src/govuk/components/footer/_footer.scss]: Ahhhhhh that makes sense. Clever! 👏

File: src/govuk/components/footer/_footer.scss (hunk 1 above)
Comment (nickcolley) [reply src/govuk/components/footer/_footer.scss]: This reads loads better thanks Chris

File: src/govuk/components/footer/_footer.scss (hunk 2 above)
Comment (nickcolley) [reply src/govuk/components/footer/_footer.scss]: I'm not a fan of clever code but feels appropriate here

Comment (nickcolley) [reply src/govuk/components/footer/_footer.scss]: I'll try and write a comment that explains this better...

Comment (ChrisBAshton) [reply src/govuk/components/footer/_footer.scss]: Thanks, this is much better!

**Comment by 36degrees** [comment]
I think there's still a couple of places where this isn't doing what we want – I think we're hardcoding some logic to try and meet a couple of specific examples (the GOV.UK Footer, three columns and one column) but there's a couple of other scenarios that are breaking the grid too.

For example, if you put the 2-column section on the right, it does not fit with the grid (you end up with a ⅖ column and a ⅗ column:

localhost_3000_components_footer_GOV UK-Flipped_preview (1)

If you have two columns, it should either go to two ½ columns, or two ⅓ columns and a third blank ⅓ column, but at the minute we get two uneven columns:

localhost_3000_components_footer_Two-equal-columns_preview

If we have only one column it kind of works as-is as it fills the entire footer, though I question whether there might be cases where a user might want to restrict it to a ⅓ column:

localhost_3000_components_footer_One-column_preview

**Comment by 36degrees** [comment]
I've currently these as examples on top of this branch – would it be useful for me to commit them and push them as something to work from?

**Comment by nickcolley** [comment]
@36degrees yes please, we may be able to make use of the column modifier classes that already exist to tighten this up even more.

**Comment by nickcolley** [comment]
Actually I've changed my mind I think those examples look how I'd expect, given the context of a fall page where most layouts are two thirds / one thirds it feels balanced. Having an empty slot feels worse?

**Comment by 36degrees** [comment]
Neither of the examples align with the grid – the first example is more like a ⅖ column and a ⅗ column, and the second example is two uneven columns.

**Comment by nickcolley** [comment]
I think I know what you're saying, I wonder if it's always been a bit off...

**Comment by nickcolley** [comment]
I've pushed up a change that uses CSS Grid instead of Flexbox.

This would mean IE11 would use the fallback layout (simple inline block that does not align with the grid).

Would also be a breaking change since I've moved the modifier...

What do you think?

- https://caniuse.com/#feat=flexbox
- https://caniuse.com/#feat=css-grid

**Comment by nickcolley** [comment]
I think we can make a small breaking change to sort this properly, so have pushed the main issues around the grid alignment back into the main issue for the next milestone: https://github.com/alphagov/govuk-frontend/issues/1539#issuecomment-527123152

I will reduce this pull request back to the original intent, so we can ship a small improvement to three column layouts only.

**Pull request merged**